packages/dates/src/utilities/formatDate.ts

```diff
@@ -1,16 +1,38 @@
+import memoize from 'lodash/memoize';
```

Comment:
No Lodash please, we have a @shopify/function-enhancers package to use instead

Reply:
I run into a problem with `quilt/packages/function-enhancers/src/memoize.ts` where it expect the enhanced function and resolver to return the same type.
The resolver returns string and the enhanced function returns Object.

I am thinking of returning string because same option may not always be same reference.

https://github.com/Shopify/quilt/pull/643/files#diff-36b8ddd38f1ae318116959216b4a4c2bR28

Reply:
@michelleyschen Is it possible to return different type between resolver and enhanced function?

Reply:
It was intended to allow this but I might have typed it wrong. 

You can updated these two line to restrict the argument of the resolver but not the return value.
https://github.com/Shopify/quilt/blob/master/packages/function-enhancers/src/memoize.ts#L7
https://github.com/Shopify/quilt/blob/use-mounted-ref/packages/decorators/src/memoize.ts#L3

Reply:
I will try to see if I have time to fix it today actually

Reply:
Thanks!

Reply:
https://github.com/Shopify/quilt/pull/685